src/controllers/DynamicFacetQueryController.ts

```diff
@@ -101,6 +108,14 @@ export class DynamicFacetQueryController {
     return this.facet.queryController.getEndpoint().search(query);
   }
 
+  protected get filterFacetCount() {
```

Comment:
This logic (and the other changes in that file) are repeated for all types of DynamicFacet.

I think it could be extracted into a `DynamicFacetRequestBuilder` (feel free to change the name :D ) module which exposes a buildRequest option to share the logic ?

Or just a module that would be in charge of determining only the `filterFacetCount` property value for each facet and could be shared between all of them.

In any case, seems like there's quite a bit of repetition between all dynamic facet query builder part of things, and it might be time to think about refactoring that a bit.

Reply:
The normal facet & ranges don't repeat the code, DynamicFacetRangeQueryController.ts  only has 1 more line which calls the same protected method. 
We can't really extract the DynamicFacet/DynamicHierarchicalFacet in a clean manner because they don't have the same interfaces, they would be a few conditions between them. (ex: delimitingCharacter) I'll try however to extract some parts of the code that are repetitive.